crates/file/src/sync_file_reader.rs

```diff
@@ -0,0 +1,29 @@
+use wasm_bindgen::{throw_str, throw_val};
+
+use wasm_bindgen::{prelude::*, JsCast, UnwrapThrowExt};
+use web_sys::FileReaderSync;
+use crate::FileReadError;
+use crate::blob::Blob;
+
+/// Asynchronously converts `blob` into a text string and then passes it to the `callback`.
+///
+/// If the returned `FileReader` is dropped before the callback is called, the read will be
+/// cancelled.
+pub fn read_as_text (blob: &Blob) -> Result<String, FileReadError>
+{
+	let fr = FileReaderSync::new().unwrap_throw();
+	fr.read_as_text(blob.as_ref()).map_err(
+		|e| {
+			throw_str(&format!("{:?}", e));
+
+			let e= e.as_string().unwrap_throw();
+			match e.as_ref () {
```

Comment:
@hamza1311
I'm having some trouble trying to log a JsValue. [`throw_str`](https://docs.rs/wasm-bindgen/0.2.74/wasm_bindgen/fn.throw_str.html) or [`throw_val`](https://docs.rs/wasm-bindgen/0.2.74/wasm_bindgen/fn.throw_val.html) from `wasm_bindgen` don't seem to be working, as browser tests output (this was interactive, as I haven't been able to get headless tests running, I'm on WSL):
```
failures:

---- web::sync_file_from_str output ----
    JS exception that was thrown:
        Error: `unwrap_throw` failed
        init/imports.wbg.__wbindgen_throw@http://127.0.0.1:8000/wasm-bindgen-test:799:15
        wasm_bindgen::throw_str::h70fa659b3cee3ef4@http://127.0.0.1:8000/wasm-bindgen-test_bg.wasm:wasm-function[2172]:0x8db5a
... really long stack trace
```
which looks to be the `unwrap_throw` on the `JsValue` passed to `map_err` by `read_as_text`.

I really just want to process the `JsValue` into a `FileReadError`, but looking at the `FileReader` implementation didn't help as it used callbacks and the exception API looked different.

Reply:
cc @hamza1311

Reply:
I'm sorry it took me so long to get back to you, been busy with things irl. I'm not really sure what's happening here. I've never experienced it. You can also use `gloo_console` to log. That would make the output look better too.

Reply:
That doesn't work. Though I think I know why the `throw_str` stuff isn't working either. Rather obvious in hindsight, `SyncFileReader` is only available in Web Workers and the context that the tests are being run in is clearly _not_ that.